# HG changeset patch
# User Mohamed Ez-zarghili <mohamed.ez.zarghili@oracle.com>
# Date 1575929296 -3600
#      Mon Dec 09 23:08:16 2019 +0100
# Node ID 66c4494134456d8649d274724c0a7679475353d2
# Parent  78d2004f65eb7f60013bfb52ddac463a332212a1
commit change

JPRT slaves provide FreeType 2.3.4 and FT_Property_Set was
only introduced in FreeType 2.4.11.

diff -r 78d2004f65eb -r 66c449413445 src/share/native/sun/font/freetypeScaler.c
--- a/src/share/native/sun/font/freetypeScaler.c	Wed Dec 04 16:24:16 2019 +0000
+++ b/src/share/native/sun/font/freetypeScaler.c	Mon Dec 09 23:08:16 2019 +0100
@@ -30,9 +30,6 @@
 #include "sun_font_FreetypeFontScaler.h"
 
 #include <stdlib.h>
-#if !defined(_WIN32) && !defined(__APPLE_)
-#include <dlfcn.h>
-#endif
 #include <math.h>
 #include "ft2build.h"
 #include FT_FREETYPE_H
@@ -42,7 +39,6 @@
 #include FT_OUTLINE_H
 #include FT_SYNTHESIS_H
 #include FT_LCD_FILTER_H
-#include FT_MODULE_H
 
 #include "fontscaler.h"
 
@@ -251,52 +247,6 @@
     }
 }
 
-typedef FT_Error (*FT_Prop_Set_Func)(FT_Library library,
-                                     const FT_String*  module_name,
-                                     const FT_String*  property_name,
-                                     const void*       value );
-
-/**
- * Prefer the older v35 freetype byte code interpreter.
- */
-static void setInterpreterVersion(FT_Library library) {
-
-    char* props = getenv("FREETYPE_PROPERTIES");
-    int version = 35;
-    const char* module = "truetype";
-    const char* property = "interpreter-version";
-
-    /* If some one is setting this, don't override it */
-    if (props != NULL && strstr(property, props)) {
-        return;
-    }
-    /*
-     * FT_Property_Set was introduced in 2.4.11.
-     * Some older supported Linux OSes may not include it so look
-     * this up dynamically.
-     * And if its not available it doesn't matter, since the reason
-     * we need it dates from 2.7.
-     * On Windows & Mac the library is always bundled so it is safe
-     * to use directly in those cases.
-     */
-#if defined(_WIN32) || defined(__APPLE__)
-    FT_Property_Set(library, module, property, (void*)(&version));
-#else
-    void *lib = dlopen("libfreetype.so", RTLD_LOCAL|RTLD_LAZY);
-    if (lib == NULL) {
-        lib = dlopen("libfreetype.so.6", RTLD_LOCAL|RTLD_LAZY);
-        if (lib == NULL) {
-            return;
-        }
-    }
-    FT_Prop_Set_Func func = (FT_Prop_Set_Func)dlsym(lib, "FT_Property_Set");
-    if (func != NULL) {
-        func(library, module, property, (void*)(&version));
-    }
-    dlclose(lib);
-#endif
-}
-
 /*
  * Class:     sun_font_FreetypeFontScaler
  * Method:    initNativeScaler
@@ -336,7 +286,6 @@
         free(scalerInfo);
         return 0;
     }
-    setInterpreterVersion(scalerInfo->library);
 
 #define TYPE1_FROM_JAVA        2
 
